PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -1265,6 +1468,20 @@ decltype(PMTraceConsumer::mPresentByThreadId.begin()) PMTraceConsumer::CreatePre
 {
     DebugCreatePresent(*newEvent);
 
+    newEvent->TrackingIndex = mCurrentPresentTrackingIndex;
+
+    if (mAllPresents[mCurrentPresentTrackingIndex] != nullptr)
+    {
+        // Existing present not completed by the time the circular buffer has come around.
+        // Remove this lost present.
+        fprintf(stderr, "Warning: present from process %u overwritten.\n", mAllPresents[mCurrentPresentTrackingIndex]->ProcessId);

```

Comment:
I think we don't always want this logging. This probably should be scoped in some way.